### DISCUSSION TITLE
Binding Async Java Functions to JavaScript

### DISCUSSION THREAD

**Discussion by webfolderio** [caoccao/Javet#71] [Q&A] [answered]
Hi,

I am trying to bind a Java method to JavaScript but couldn't figure out how to use it with Promise.
I check the documentation and [promise test class](https://github.com/caoccao/Javet/blob/main/src/test/java/com/caoccao/javet/values/reference/TestV8ValuePromise.java) but still couldn't find the solution.

Thanks

Example usage:

Java:
```java
class Service {
 public String longRunningTask() {
  return "done";
 }
}
```

JavaScript:
```js
var myFunc = async function() {
  var service = new Service();
  await service.longRunningTask();
};
```

_Note: I am using pure v8 without node support._

**Comment by caoccao** [comment]
Nice catch. That piece of work hasn't been fully implemented, as I'm adding the Proxy based injection now. Promise is a wonderful feature that requires certain amount of work. You are welcome to discord (link is at home page), discussing that with me in detail.

**Reply by caoccao** [reply]
The gap is being closed now. Please check this [commit](https://github.com/caoccao/Javet/commit/84fa2c28b09b426ad284074dbc52dab7abc3ceef) out for detail and let me know if you have any questions.

**Comment by caoccao** [comment, marked as answer]
Please check out the source code of [Javenode](https://github.com/caoccao/Javenode) that has many practical usage of `Promise` for async operations.
